readthedocs/projects/tasks.py

```diff
@@ -907,20 +930,41 @@ def sync_files(
     version = Version.objects.get_object_or_log(pk=version_pk)
     if not version:
         return
-    if not pdf:
-        remove_dirs([
-            version.project.get_production_media_path(
-                type_='pdf',
-                version_slug=version.slug,
-            ),
-        ])
-    if not epub:
-        remove_dirs([
-            version.project.get_production_media_path(
-                type_='epub',
-                version_slug=version.slug,
-            ),
-        ])
+    if delete:
```

Comment:
I think there's an issue around this. While I was testing the removal feature, this flag was always `False`. I'm actually not 100% sure why we need this flag at all. Wouldn't we always want to remove old PDFs/ePubs?

Reply:
It should only be true when being called w/ the broadcast. I guess we could delete the old ones before replacing them with the new ones, but that seems like wasted effort, but arguably could be done.

Reply:
> I guess we could delete the old ones before replacing them with the new ones, but that seems like wasted effort, but arguably could be done.

For storage, you have to do this. There isn't an "overwrite" option.

Reply:
> It should only be true when being called w/ the broadcast.

Is the implication here that this won't happen when run locally? I wasn't able to successfully test deleting artifacts when PDF/Epubs are disabled locally.